### PR TITLE
Network: Adds OVN address forwards

### PR DESCRIPTION
**Pull request by tomponline**

### CODE REVIEW

**Comment by tomponline** [comment]
@stgraber aside from the changes discussed here https://discuss.linuxcontainers.org/t/lxd-floating-ip-addresses/11801/43?u=tomp this will soon be finished, so if you have chance to take a quick first-pass look before its finalised that would be appreciated.

**Comment by tomponline** [comment]
@stgraber I ran out of time to add docs but will add tomorrow, other than that its ready for review.

**Comment by tomponline** [comment]
@stgraber added docs now, ready for review.

File: client/interfaces.go [hunk 1]
```diff
@@ -226,6 +226,13 @@ type InstanceServer interface {
 	RenameNetwork(name string, network api.NetworkPost) (err error)
 	DeleteNetwork(name string) (err error)
 
+	// Network forward functions ("network_forward" API extension)
+	GetNetworkForwards(networkName string) ([]api.NetworkForward, error)
```
Comment (stgraber) [inline client/interfaces.go:230]: We probably should have a `GetNetworkForwardAddresses(networkName string) ([]string, error)` to line up with our usual pattern

File: shared/api/network_forward.go [hunk 2]
```diff
@@ -91,7 +91,7 @@ type NetworkForwardPut struct {
 	// Example: My public IP forward
 	Description string `json:"description" yaml:"description"`
 
-	// Forward configuration map (refer to doc/networks.md)
+	// Forward configuration map (refer to doc/network-forwards.md)
```
Comment (stgraber) [inline shared/api/network_forward.go]: Can you rebase that one into the commit which introduces the API?

File: doc/networks.md [hunk 3]
```diff
@@ -355,6 +355,17 @@ lxc ls
...
 
+Network forwards:
+
+OVN networks support [network forwards](network-forwards.md).
+
+The allowed listen addresses are those that are defined in the uplink network's `ipv{n}.routes` settings, and the
+project's `restricted.networks.subnets` setting (if set).
+
+The listen address used cannot overlap with a subnet that is in use with another network.
+
```
Comment (stgraber) [inline doc/networks.md:366]: Feels a bit weird to have this be in `networks.md`. Given that the only two types that can actually handle forwards are bridged and ovn, I think it'd probably be best to cover their specific restrictions directly in the forwards doc.

File: shared/api/network_forward.go (hunk 2 above)
Comment (tomponline) [reply shared/api/network_forward.go]: Yep that was an oversight, hence the short commit message.

File: doc/networks.md (hunk 3 above)
Comment (tomponline) [reply doc/networks.md:366]: OK will move, but leave the link there to the forwards section.

File: client/interfaces.go (hunk 1 above)
Comment (tomponline) [reply client/interfaces.go:231]: Yep I've added.

**Comment by tomponline** [comment]
@stgraber ready for review again

**Comment by tomponline** [comment]
@stgraber rebased, also now im onto adding network forwards to bridge networks, and whilst reviewing your BGP branch, it occurred to me that network forwards might actually need to be node-specific (at least for bridge networks), what do you think (and if so, can you hold off merging the ovn branch so I can apply the schema changes needed)?

**Pull request merged**